include/stdx/tuple_algorithms.hpp

```diff
@@ -319,16 +319,17 @@ template <tuplelike... Ts> constexpr auto cartesian_product_copy(Ts &&...ts) {
         return []<typename First, typename... Rest>(First &&first,
                                                     Rest &&...rest) {
             auto const c = cartesian_product_copy(std::forward<Rest>(rest)...);
-            return std::forward<First>(first).apply(
-                [&]<typename... Elems>(Elems &&...elems) {
-                    auto const prepend = [&]<typename E>(E &&e) {
-                        return c.apply([&](auto... subs) {
-                            return make_tuple(tuple_cat(
-                                make_tuple(std::forward<E>(e)), subs)...);
-                        });
-                    };
-                    return tuple_cat(prepend(std::forward<Elems>(elems))...);
-                });
+            return std::forward<First>(first).apply([&]<typename... Elems>(
+                                                        Elems &&...elems) {
+                // NOTE: it's always used, but gcc sometimes gets it wrong :(
```

Comment:
It's not used if the pack is empty on line 331. That's why GCC complains sometimes.

Reply:
Maybe I can update the note to:

```c++
                // NOTE: If the `elems` pack is empty, then GCC warns that
                //       `prepend` is not used (since it won't be called)
```

...or remove it.